mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala

```diff
@@ -113,6 +113,13 @@ class VectorsSuite extends SparkFunSuite with Logging {
     assert(vec.toArray === arr)
   }
 
+  test("zero-length sparse vector") {
```

Comment:
While we're doing this we may as well also add a test to `intercept` the exception for negative size (as per the other sparse vector construction tests), for both `ml` and `mllib`